### ISSUE TITLE
kaizen sso login failures

### ISSUE THREAD

**Issue by larsks** [CCI-MOC/ops-issues#9]
Users are reporting problems logging into kaizen with institutional accounts. The initial login will result in a failure message, but attempting to visit https://kaizen.massopen.cloud after that will work correctly.

**Comment by larsks** [comment]
Looked at this with @knikolla  this afternoon. We are seeing an invalid redirect URL that looks like:

```
https://kaizen.massopen.cloud:13000, 13000/v3/auth/OS-FEDERATION/identity_providers/moc/protocols/openid/websso?origin=https://kaizen.massopen.cloud/dashboard/auth/websso/', referer: https://kaizen.massopen.cloud/
```

Look at the `13000, 13000` there in the netloc part of the URL.

I built a debug version of `mod_auth_openidc` and it looks like the bad value is clearly coming from the `X-Forwarded-For` header.  This line (https://github.com/larsks/mod_auth_openidc/blob/v1.8.8-patches/src/util.c#L323) produces this output: 

```
oidc_get_current_url_port: port_str from x-forwarded-port=|13000, 13000|
```

**Comment by larsks** [comment]
Problems with x-forwarded-port point to haproxy. I'll see if I can figure out some useful diagnostics on that end maybe on Monday.

**Comment by larsks** [comment]
I figured it out.

Previously, there was an issue in the haproxy configuration created by Director; it was not setting the `X-Forwarded-For` header, which resulted in `mod_auth_oidc` generating invalid redirects (see rhbz-1810213 for details).

This bug has been fixed, but as a workaround to the previous problem, our Keystone Apache configuration included:

```
RequestHeader append X-Forwarded-Port 13000 early
```

Now that haproxy is correctly setting this header, we're appending a *second* value, which results in the header going from `X-Forwarded-Port: 13000` to `X-Forwarded-Port: 13000, 13000`.

So we just need to remove our workaround now that the issue has been resolved in Director.
